app/models/manageiq/providers/vmware/cloud_manager/vm/operations.rb

```diff
@@ -1,3 +1,14 @@
 module ManageIQ::Providers::Vmware::CloudManager::Vm::Operations
+  extend ActiveSupport::Concern
+
   include_concern 'Power'
+
+  def raw_destroy
+    raise "VM has no #{ui_lookup(:table => "ext_management_systems")}, unable to destroy VM" unless ext_management_system
+    with_provider_object(&:undeploy)
+    ext_management_system.with_provider_connection do |service|
```

Comment:
Since `with_provider_object` calls `with_provider_connection` can you do the `:undeploy` in this block so you only have to connect once?

Reply:
@VojkoR I believe what @agrare is suggesting is not to use the `with_provider_object` to undeploy but rather use it like this

```
ext_management_system.with_provider_connection do |service|
  service.post_undeploy_vapp(ems_ref)
  service.delete_vapp(ems_ref)
end
```

https://github.com/xlab-si/fog-vcloud-director/blob/master/lib/fog/vcloud_director/requests/compute/post_undeploy_vapp.rb#L36